### DISCUSSION TITLE
Alert component updates

### DISCUSSION THREAD

**Discussion by Richard592** [UKHomeOffice/design-system#358] [General]
Hi, the Alert component uses non-standard colours which need adding to the component details somewhere, They are:

.panel-border-wide.alert-default {
  border-color: #2b8cc4;
  background-color: #dbeff9;
}
.panel-border-wide.alert-success {
  border-color: #28a197;
  background-color: #c6ece9;
}
.panel-border-wide.alert-error {
  border-color: #c42b2b;
  background-color: #f3dede;
}

Thanks.

**Comment by Sarraounia** [comment]
@Richard592 do you know if this was ever added to the design system? I want to add some new text to the Alert component to help content designers write content for alerts. If your suggestion hasn't been added yet, probably makes sense to put both these updates through together.

**Comment by Richard592** [comment]
@Sarraounia it looks like it hasn't been implemented, just judging by it still being unassigned. Happy for you to add additional criteria to this open issue. How it gets prioritised I don't know though. If you need it prioritising maybe include a nudge to Laurence Piercy as he is fielding GitHub contributions (I think).

**Comment by Sarraounia** [comment]
@Richard592 I've got my content ready now for the Alert page. I am on the design system working group so will liaise with Laurence and the team about this update. 

What are your thoughts on how to communicate the non-standard colours? I have an idea but it's probably best if we have a quick chat about it. I can't tell who you are from your GitHub username so if you can write to me on Slack or send me an email, that would be great!

**Comment by Richard592** [comment]
Hi Ounie!!

Turns out it is me that contributed to the Alert component on GitHub. I haven't been receiving any updates from  gitHub, until two minutes ago.

Laurence is still the best person to contact about it though. But happy to have a chat about it to see if i can help at all.


Richard Chilver

Senior Service Designer

Migration and Borders Technology

Home Office Data, Digital and Technology

Sheffield Delivery Centre<https://collaboration.homeoffice.gov.uk/display/DH/Sheffield+Delivery+Centre>

Vulcan House Steel

4-6 Millsands

Sheffield

S3 8NU

HO Digital Data and Technology Strategy<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.gov.uk%2Fgovernment%2Fpublications%2Fhome-office-digital-data-and-technology-strategy-2024%2Fhome-office-digital-data-and-technology-strategy-2024&data=04%7C01%7CRichard.Chilver%40digital.homeoffice.gov.uk%7Ca08dfbb7e35e472f8afe08d9469c1b97%7C90a6982e16354aa087a6d5d63c7bc66e%7C0%7C0%7C637618458554297059%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=hInrXZfERg0DOYo69KO1nIwN8rhwOndHJeAKfXQYxs8%3D&reserved=0>

________________________________

**Comment by anandamaryon-gov** [comment]
Just to loop into this issue to mention that the current colours used on the alert backgrounds don't allow enough contrast when using links within alerts. So if the colours are being looked at, this would be good to consider.

**Reply by Sarraounia** [reply]
Thanks @anandamaryon-gov there's going to be a workshop soon around colours in the design system soon so we can look at this then.

**Comment by Sarraounia** [comment]
I have drafted some content to add to the current [Alerts](https://design.homeoffice.gov.uk/components?name=Alert) component about how to write for it, and would like feedback from others across the Home Office.

PLEASE ADD ANY COMMENTS BY 13 MAY 2022.


##Writing for this component 

Try to keep alerts to one or two lines of text and not necessarily in full sentences, given the lack of space. If the text does not end in a link and is a full sentence, include a full stop.  

Tell users what to do next. If you can, give them a link that will take them directly to further support or the next action. 

Only use bold for headings. 

If the alert is about people, make sure your content is dynamic and personalised. Use ‘person’ when it’s about one person and ‘people’ for more than one and include names if you can. 


###Multiple alerts 

Avoid using lots of alerts next to each other and put all the information in one alert where possible. 

Some journeys may need multiple alerts on a single screen. If, for example, the journey has one successful and one error alert, they should be in priority order. 

###Alert placement

Although alerts are usually put at the top of the screen, this is not always the case. Where you put the alert should suit the needs of your service.

**Comment by Richard592** [comment]
Hi Ounie,

In the part about dynamic content, is it worth expanding on that to include personalisation? Not sure what the content policy is on that and whether we will ever know users actual name. I'm thinking about it from the 'you' 'we' 'your' perspective as well.

Being picky here, but would changing 'various' to 'multiple' make the line under the Multiple alerts heading easier to interpret?

Maybe the last two points could go under a 'Positioning' header so they don't feel like they belong to the multiple alerts content. I know there's overlap there but might be clearer to have a positioning section as well as I know from experience that has not been clear in the past.

**Reply by Sarraounia** [reply]
Thanks for your comments @Richard592. Good point about the dynamic content. It is good practice to use names. We've done this in other services so have updated the guidance above. Didn't include anything about addressing users ('you' 'we' etc) as that's already covered in the GOV.UK style guide and we want people to follow that in the first instance. 

The content was 2i'ed and that was a suggestion I got to use both 'various' and 'multiple.' I don't feel strongly about it either way. Did you find it difficult to understand as it currently is?

Have updated the content now to include a new header for 'Alert placement.' They do overlap a bit, especially the point about the priority order. Think that fits better in 'Multiple alerts' heading. What do you think?

**Comment by Richard592** [comment]
They're all good updates to it. The additional 'alert placement' helps clarify things, even if there's an overlap.
I did struggle with the 'various' and 'multiple' being interchangeable, but that's probably just my poor grasp of the English language. My interpretation is that 'various' means different types of alerts that are a specific list that isn't shown here, so where do I find that out? Whereas 'multiple' is directly stating more than one of any type. But I'm not going to die on a hill for a change or anything!

**Reply by Sarraounia** [reply]
Oh I see what you mean! It's really helpful feedback and exactly the challenge my content brain needs on a Friday afternoon. I've given it another stab now - hopefully it's a bit clearer..

**Comment by OscarJacksonHO** [comment]
Hi everyone, 

Off the back of a product alignment workshop, I’ve found that there might be some improvements we could make on our current approach to [alerts](https://design.homeoffice.gov.uk/components?name=Alert).

I thought I’d put some thoughts together before digging into this too much to get the conversation going. 

Please share any thoughts.

Potential areas for improvement:

• making the purpose of different alerts more intuitive and distinct
• renaming ‘alerts’
• removing the ‘alert indicating an error’


### Making the purpose of different alerts more intuitive and distinct

**What**

1. Explore adding some icons to the alerts, especially the ‘alert notifying of a successful outcome’. Perhaps, something like what ONS have done with theirs:  [ONS panels](https://service-manual.ons.gov.uk/design-system/components/success-panel )

2. Define the use of each component and think about splitting the guidance out so that we can provide something more comprehensive and distinct for each variant.

3. Remove headings from the examples for ‘standard alerts’ to discourage their use (maintain this as an option)



**Why**

1. Currently our ‘alerts’ are only distinguishable from each other by their colour. Personally, I think the colour of some of these doesn’t give the user a recognisable hint as to their purpose. Icon use can also have accessibility benefits: [Making Content Usable for People with Cognitive and Learning Disabilities (w3.org)](https://www.w3.org/TR/coga-usable/#use-icons-that-help-the-user-pattern)


2. We don’t have much guidance on what for/when to use alerts. Currently, the purpose of our ‘standard alert’ is to ‘alert users to important information’. This is very broad and leaves the user to decide what ‘important information’ is, which is likely to result in inconsistency.


3. If the information displayed in standard panels is about the page, it should be under the h1 and, otherwise, the information should be related enough to the page content to sit directly under another meaningful heading that’s already on the page.



### Renaming ‘alerts’

**What**
Rename the ‘alerts’.

Some options could be:

-	panel (information panel, success panel etc.)
-	blockquote? (I don’t particularly like this one)

Welcoming thoughts on this, so feel free to reply with any other suggestions

**Why**
I don’t think that the name ‘alert’ truly speaks to the purpose of these components. This implies danger which isn’t always the consequence of missing the information.

The current names (e.g. alert indicating an error) are also not frontloaded and are longer than they could be.


### Removing the ‘alert indicating an error’

**What**
Remove the ‘alert indicating an error’.
Explore adding a ‘[warning panel](https://service-manual.ons.gov.uk/design-system/components/warning-panels)’. For example, to house the “this action cannot be undone” text on a deletion page.

**Why**
I think this variant has a few flaws:

• it’s too similar to the components that make up the [recover from validation errors](https://design-system.service.gov.uk/patterns/validation/) GOV pattern
• if a user is unable to complete a task we should either use the above pattern or a [there is a problem with the service](https://design-system.service.gov.uk/patterns/problem-with-the-service-pages/) page

**Reply by benjest85** [reply]
@OscarJacksonHO  We should get our heads together on this as I've just looked at the Alert component.

**Comment by benjest85** [comment]
After discussing, we agreed that for "What" 
1. Explore the value of adding these. The ONS example just includes one icon for success and no others so would need to look at what other icons would need adding, if any. 
2. Agree this needs doing. It's worth trying to explore where these alerts are used, the context and use cases. Aim would be to try and make the use of these more explicit rather than generic "we need to tell the user something"
3. Tied to number 2

In addition to Oscar exploring the points above. I will look at the colour as part of the colour palette work I plan to do. Agree that colours like the "turquoise" in the success alert feel "off brand". 

Renaming alerts also feels tied to the research above. I agree that this name doesn't suit the intention.

**Reply by Daniele1811** [reply]
@benjest85  @Sarraounia

I am currently working on alert outcomes for different edge cases of assigning units/tasks to officers:

- Successfully assigned 1 or all units selected: green alert, tell user units were assigned.
<img width="817" height="110" alt="image" src="https://github.com/user-attachments/assets/3920c015-6b39-4997-8f5d-e5de557c860e" />

- Unable to assign these units/tasks (as they have just been assigned by someone else while user is performing the task): MOJ guidance suggests we should use a red error alert here when something has changed significantly, for example a case was deleted or assigned to someone else. HO guidance says to use the red version of the alert to confirm that something has gone wrong.
- Red alert: error - something went wrong on the back end that prevents the user from completing the task.
<img width="634" height="87" alt="image" src="https://github.com/user-attachments/assets/87c0b2b3-8949-40b8-b624-e7d0e3abebdd" />

- Partial success: I have seen on [the HODS](https://design.homeoffice.gov.uk/design-system/components?name=Alert) that it says 'Some journeys may need multiple alerts on a single screen. If, for example, the journey has one successful and one error alert, they should be in priority order.' Similarly, on [MOJ DS](https://design-patterns.service.justice.gov.uk/components/alert/), it says 'An error and success alert under an H1 in a case management system shows a user that one of their tasks failed, and another was successful.' This is interesting because they then say 'that the ‘alert’ read out to screenreader users should only be applied to one alert.'

Following a conversation with one of the devs, we clarified why we should only use one alert for a partial success:

- Otherwise we have no way to focus both (it wouldn't be accessible). 
- Overall, the user goal is to make sure the unassigned units are assigned (not left without an assignee). So even if someone else takes one or two tasks, its not a failure so not an error to use a red alert.
- We then concluded that a partial outcome should be communicated on a green alert with a sentence under the success message (on the alert) that 'You cannot assign unit/task X, as it has already been assigned to someone else. 
<img width="1111" height="207" alt="image" src="https://github.com/user-attachments/assets/e8eaec20-cce8-43dc-99ea-5a1cb169ce3a" />

We also decided to use a blue alert for when users are unable to assign units/tasks, as it is not an error and is simply communicating an update. 
<img width="938" height="111" alt="image" src="https://github.com/user-attachments/assets/736a1e2a-22fa-46c7-b7ee-361beeb3ce27" />

Feedback would be appreciated please. :-)

**Reply by benjest85** [reply]
Hey @Daniele1811 . Sounds reasonable to me. You're trying to do something a little more nuanced than the standard cases these alerts were probably intended for: "A thing worked, a thing didn't work" etc. You're trying to create a bit of a hybrid of these components it seems. 

Given these are used mainly internally, I think we have a little more flexibility on how they are used. Is there a case for a yellow/amber one that covers your scenario of "something worked but with caveat"? Maybe overkill given we could carry on forever with more colours for a vast numbers of random scenarios.
